packages/stripe-graphql/src/schema/stripe.ts

```diff
@@ -0,0 +1,39 @@
+import { GraphQLYogaError } from '@graphql-yoga/node'
+
+import { builder } from '../builder'
+import { isAllowed } from '../utils'
+
+builder.objectType('Stripe', {
+  fields: (t) => ({
+    customer: t.field({
+      type: 'StripeCustomer',
+      args: {
+        id: t.arg.string({
+          required: true
+        })
+      },
+      resolve: async (_parent, { id }, context) => {
+        if (!isAllowed(id, context)) {
+          throw new GraphQLYogaError('user is not allowed to see info from this stripe id')
+        }
+
+        const { stripe } = context
+
+        const customer = await stripe.customers.retrieve(id)
+
+        if (customer.deleted) {
+          throw new GraphQLYogaError('customer is deleted')
+        }
+
+        return customer
+      }
+    })
+  })
+})
+
+builder.queryFields((t) => ({
+  stripe: t.field({
+    type: 'Stripe',
+    resolve: () => ({} as any)
```

Comment:
looks like you are defining the backing type for the 'Stripe' object in the builder as having a customer property, removing that customer property should allow you to avoid the any here.  The type defined in the builder is used for the expected return type of a resolve, and for the parent argument in resolvers for that type, but does not constrain what fields you can defined (you can still define your customer field on `Stripe` even if the backing typescript type does not have it.

Reply:
Could you guide me on how to do this with Pothos?

I want the `customer` (and later other fields such as `customers`, `paymentMethods`, `plans` etc.) to be scoped under `stripe` like this:

```graphql
query {
  stripe {
    customer(id: "cus_xxx") {
      id
      name
    }
    plans {
      id
      name
    }
  }
}
```

Reply:
I think you have it basically right, you just need to delete line 12 of builder.ts

Those type mappings are basically telling Pothos what data represents the corresponding graphql object.   In this case you are saying a stripe object is represented by an object with a customer property, and here you are returning an object without that property.   You don't actually need (or want) that customer property as part of the parent objects in your resolvers on the Stripe object, so you can just remove that from the type definition in builder.ts